### PR TITLE
UF-320: Including workbench and project preferences

### PR DESCRIPTION
**Pull request by paulovmr**
* Configuring preferences API for the workbench.
* Project preferences: specific preferences for each project.

### CODE REVIEW

**Comment by paulovmr** [comment]
Part of an ensemble:

* https://github.com/uberfire/uberfire/pull/558
* https://github.com/droolsjbpm/guvnor/pull/386
* https://github.com/droolsjbpm/kie-wb-common/pull/558
* https://github.com/droolsjbpm/kie-wb-distributions/pull/404

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
I'm going to make some changes on this PR, please wait for review/merge.

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/admin/DefaultAdminPageHelper.java [hunk 1]
```diff
@@ -96,6 +96,12 @@ public void setup() {
                                    command.execute( response.getTotal() );
                                }
                            }, ( o, throwable ) -> false ).search( new SearchRequestImpl( "", 1, 1, null ) ) );
+
+        adminPage.addPreference( "root",
+                                 "ProjectPreferences",
+                                 "Project",
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/admin/DefaultAdminPageHelper.java]: i18n

File: kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/preferences/config/WorkbenchPreferenceScopeTypes.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        validateType( type );
+
+        return defaultKeyByType.get( type ) == null;
+    }
+
+    @Override
+    public String getDefaultKeyFor( final String type ) throws InvalidPreferenceScopeException {
+        validateType( type );
+
+        final DefaultKey defaultKey = defaultKeyByType.get( type );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/preferences/config/WorkbenchPreferenceScopeTypes.java]: Do we need to observe ```ProjectContextChangeEvent``` and add the Project's "Encoded Identifier" in the ```defaultKeyByType``` for ```WorkbenchPreferenceScopes.PROJECT```?

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/admin/DefaultAdminPageHelper.java (hunk 1 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/admin/DefaultAdminPageHelper.java]: Done

File: kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/preferences/config/WorkbenchPreferenceScopeTypes.java (hunk 2 above)
Comment (paulovmr) [inline kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/preferences/config/WorkbenchPreferenceScopeTypes.java]: @manstis That is not necessary. The default key should only be set when the same key is used at all times for one user, which is not the case for the project scope. In this case, we should always provide one when calling the api with this scope belonging to the scope resolution strategy.

**Comment by paulovmr** [comment]
jenkins retest this

**Pull request merged**